### DISCUSSION TITLE
Programmatically validate icon layout and color choice

### DISCUSSION THREAD

**Discussion by JayKayAce** [ni/labview-icon-editor#204] [General]
**User story:**
As a VI Package Maintainer of an Open Source project, I would like an API to programmatically validate that an icon follows my icon design so that new contributions to my project are consistent with the existing Icons. 

The usage scenario is that the Package developer can create unit tests and validate that the icons of all new functions developed in the project follow the coding standards.

The validation scenarios could be:
- Ensure that a certain template is used based on the VI name
- Ensure no overlapping glyphs
- Test if the Library layer has been activated or deactivated

**Comment by svelderrainruiz** [comment]
Some of this functionality has been done by me on the feature that detects mismatches on owning banners. [here](https://github.com/ni/labview-icon-editor/issues/40)

**Comment by svelderrainruiz** [comment]
We will discuss implementation first before making this an issue.
